scripts/tasks/generate-license-files.mjs

```diff
@@ -23,24 +24,34 @@ async function exists(filename) {
     }
 }
 
+const licenses = JSON.parse(
+    spawnSync('pnpm', ['licenses', 'list', '--json'], { encoding: 'utf-8' }).stdout
+);
+
+if (!licenses || typeof licenses !== 'object') {
+    throw new Error('Failed to list dependencies');
+}
+
+const list = Object.values(licenses).flat();
+
 async function findLicenseText(depName) {
     // Iterate through possible names for the license file
     const names = ['LICENSE', 'LICENSE.md', 'LICENSE.txt'];
 
     // We would use require.resolve, but 1) that doesn't work if the module lacks a "main" in its `package.json`,
     // and 2) it gives us a deep `./path/to/index.js` which makes it harder to find a top-level LICENSE file. So
     // just assume that our deps are hoisted to the top-level `node_modules`.
-    const resolvedDepPath = path.join(process.cwd(), 'node_modules', depName);
+    const resolvedDepPath = list.find((dep) => dep.name === depName).paths[0];
 
     for (const name of names) {
-        const fullFilePath = path.join(resolvedDepPath, name);
+        const fullFilePath = join(resolvedDepPath, name);
```

Comment:
Converted to `.mjs` hoping `import.meta.resolve` would work, but that would need [shamefully-host](https://pnpm.io/npmrc#shamefully-hoist), which I think might be aptly named.

I shamefully put this hack together instead.

Reply:
I was checking if import.meta.resolve would somehow work for this case, which it didn't, but I kept it since I find esm syntax nicer to work with. I can revert it back to .js.